routing/world_graph.cpp

```diff
@@ -0,0 +1,37 @@
+#include "routing/world_graph.hpp"
+
+namespace routing
+{
+using namespace std;
+
+WorldGraph::WorldGraph(unique_ptr<CrossMwmIndexGraph> crossMwmGraph,
+                       unique_ptr<IndexGraphLoader> loader, shared_ptr<EdgeEstimator> estimator)
+  : m_crossMwmGraph(move(crossMwmGraph)), m_loader(move(loader)), m_estimator(estimator)
+{
+  CHECK(m_loader, ());
+  CHECK(m_estimator, ());
+}
+
+void WorldGraph::GetEdgeList(Segment const & segment, bool isOutgoing, vector<SegmentEdge> & edges)
+{
+  IndexGraph & indexGraph = GetIndexGraph(segment.GetMwmId());
+  indexGraph.GetEdgeList(segment, isOutgoing, edges);
+
+  // TODO remove this return to activate m_crossMwmGraph.
+  return;
+
+  if (!m_crossMwmGraph || !m_crossMwmGraph->IsTransition(segment, isOutgoing))
+    return;
+
+  ASSERT(m_twins.empty(), ());
+  m_crossMwmGraph->GetTwins(segment, m_twins);
+  for (Segment const & twin : m_twins)
+    edges.emplace_back(twin, 0.0 /* weight */);
+  m_twins.clear();
```

Comment:
> Is it possible to transform GetTwins() to something like

Yes, of course. It's quite possible. To do that it's necessary to change CrossMwmIndexGraph. The question was if is it possible to have several twins according to OSRM cross mwm routing. I investigated it. Yes, it happens sometimes. And for most cases it's errors. As far as I'm aware OSRM routing checks if a node id is a twin by coords. Node Ids considered as equal if the distance between them if 0.001 degrees :)